### PR TITLE
LPP-43049 Deactivated users are counted in Roles page

### PR DESCRIPTION
**Pull request by thuycao-gs**

### CODE REVIEW

File: modules/apps/users-admin/users-admin-web/src/main/java/com/liferay/users/admin/web/internal/portlet/action/EditUserMVCActionCommand.java [hunk 1]
```diff
@@ -180,6 +181,9 @@ protected void deleteUsers(ActionRequest actionRequest) throws Exception {
 
 					if (cmd.equals(Constants.DEACTIVATE)) {
 						status = WorkflowConstants.STATUS_INACTIVE;
+
+						_userGroupRoleLocalService.deleteUserGroupRolesByUserId(deleteUserId);
```
Comment (huynguyen-codeengine) [inline modules/apps/users-admin/users-admin-web/src/main/java/com/liferay/users/admin/web/internal/portlet/action/EditUserMVCActionCommand.java:185]: Delete user roles is not a good way. Re-active deactivated user will be lost all roles.

Comment (thuycao-gs) [reply modules/apps/users-admin/users-admin-web/src/main/java/com/liferay/users/admin/web/internal/portlet/action/EditUserMVCActionCommand.java]: Hi @huynguyen-codeengine 
I updated my solution.
Just check the getAssigneesTotal() function. if a user is inactive will be not counted.
Please consider and let me know if it has any?

File: portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java [hunk 2]
```diff
@@ -2118,6 +2109,24 @@ protected void validate(
 		}
 	}
 
+	private int _getAssigneesTotalWithActiveUser(
+			List<UserGroupRole> userGroupRoles)
+		throws NoSuchUserException {
+
+		int assigneesTotal = 0;
+
+		for (UserGroupRole userGroupRole : userGroupRoles) {
```
Comment (huynguyen-codeengine) [inline portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java:2118]: this solution has performance issue by getting all user one by one.

Comment (thuycao-gs) [reply portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java:2118]: It just gets users with same roleId

**Comment by huynguyen-codeengine** [comment]
Fail source format
OpenJDK 64-Bit Server VM warning: ignoring option MaxPermSize=2048m; support was removed in 8.0
Buildfile: /media/ces-user/USBDISK/liferay-portal/LPP-43049-2/portal-impl/build.xml
   [delete] Deleting: /media/ces-user/USBDISK/liferay-portal/LPP-43049-2/null1062870159.properties

format-source-latest-author:
   [delete] Deleting: /media/ces-user/USBDISK/liferay-portal/LPP-43049-2/null1062870159.properties

format-source-files:
     [java] Loading file:/media/ces-user/USBDISK/liferay-portal/LPP-43049-2/portal-impl/classes/system.properties
     [java] SLF4J: Class path contains multiple SLF4J bindings./media/ces-user/USBDISK/liferay-portal/LPP-43049-2/portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java (JavaParser)
     [java] 
     [java] SLF4J: Found binding in [file:/media/ces-user/USBDISK/liferay-portal/LPP-43049-2/util-slf4j/classes/org/slf4j/impl/StaticLoggerBinder.class]
     [java] /media/ces-user/USBDISK/liferay-portal/LPP-43049-2/portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java (SourceCheck:JavaTermOrderCheck)
     [java] SLF4J: Found binding in [jar:file:/media/ces-user/USBDISK/liferay-portal/LPP-43049-2/lib/development/slf4j-simple.jar!/org/slf4j/impl/StaticLoggerBinder.class]
     [java] SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
     [java] SLF4J: Actual binding is of type [com.liferay.util.slf4j.LiferayLoggerFactory]
     [java] Avoid chaining on method 'forName', see https://github.com/liferay/liferay-portal/blob/master/modules/util/source-formatter/src/main/resources/documentation/checks/chaining_check.markdown: ./portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java 658 (Checkstyle:ChainingCheck)
     [java] java.lang.Exception: Found 1 formatting issues:
     [java] 1: Avoid chaining on method 'forName', see https://github.com/liferay/liferay-portal/blob/master/modules/util/source-formatter/src/main/resources/documentation/checks/chaining_check.markdown: ./portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java 658 (Checkstyle:ChainingCheck)
     [java] 
     [java] 	at com.liferay.source.formatter.SourceFormatter.format(SourceFormatter.java:458)
     [java] 	at com.liferay.source.formatter.SourceFormatter.main(SourceFormatter.java:287)

BUILD FAILED
/media/ces-user/USBDISK/liferay-portal/LPP-43049-2/portal-impl/build.xml:762: The following error occurred while executing this line:
/media/ces-user/USBDISK/liferay-portal/LPP-43049-2/portal-impl/build.xml:689: Java returned: 1

**Comment by thuycao-gs** [comment]
Hi @huynguyen-codeengine ,
I have updated it. Thanks in advance.

**Comment by huynguyen-codeengine** [comment]
Forwarded at https://github.com/liferay-ts-reviewer/liferay-portal/pull/10

**Pull request closed without merging**